sentry-opentelemetry/sentry-opentelemetry-core/build.gradle.kts

```diff
@@ -21,7 +21,7 @@ tasks.withType<KotlinCompile>().configureEach {
 dependencies {
     compileOnly(projects.sentry)
 
-    compileOnly(Config.Libs.OpenTelemetry.otelSdk)
+    implementation(Config.Libs.OpenTelemetry.otelSdk)
```

Comment:
hm, why did we change it to `implementation`? It's not possible to use this module without otel, right? Or is the assumption that people will just include our `sentry-opentelemetry-core` package, but not otel itself?

Reply:
The intention was to make it easier to use `sentry-opentelemetry-agent` with `SENTRY_AUTO_INIT=false`. Currently you need to add `sentry-opentelemetry-core` to the dependencies so you can enjoy linking of errors to transactions. With `compileOnly` you'd also have to include the correct version of the OTEL SDK, with `implementation` it just drags it in. For `sentry-opentelemetry-agent` we exclude it again, which users also have to do, if they don't want to use the agent but  have a different OTEL version than we use or their build system can't figure out versions.

Another approach (for later) would be to have yet another gradle module, that we can then add to the bootstrap classloader in the agent, thus not requiring the explicit dependency on `sentry-opentelemetry-core` at all, when setting `SENTRY_AUTO_INIT=false`.

Reply:
alright, just beware that `implementation` will override whatever version the users are using (if it's lower than ours). If that's what we want, then I'm good 👍